utils/TimelockHelpers.ts

```diff
@@ -0,0 +1,98 @@
+import { BigNumber, ContractReceipt } from "ethers";
+import { AbiCoder, ParamType, Result } from "ethers/lib/utils";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+import { Timelock__factory } from "../typechain";
+
+export type QueuedTransaction = {
+  recipient: string;
+  value: number;
+  data: string;
+};
+
+export enum TransactionStatus {
+  NonExistent,
+  Queued,
+  Approved
+}
+
+export function decodeQueueTransactionsBatchParams(decodedTxData: Result) {
+  const decodedQueuedTransactions = decodedTxData.transactions as QueuedTransaction[];
+  const decodedExecutionStartTimestamp = decodedTxData.executionStartTimestamp as BigNumber;
+
+  return { queuedTransactions: decodedQueuedTransactions, executionStartTimestamp: decodedExecutionStartTimestamp };
+}
+
+export async function decodeTimelockQueuedTransactions(hre: HardhatRuntimeEnvironment, txHash: string) {
+  const txData = (await hre.ethers.provider.getTransaction(txHash)).data;
+  const decodedTxData = await Timelock__factory.createInterface().decodeFunctionData("queueTransactionsBatch", txData);
+  const decodedQueuedTransactions = decodedTxData.transactions as QueuedTransaction[];
+  const decodEdexecutionStartTimestamp = decodedTxData.executionStartTimestamp as BigNumber;
+
+  return { queuedTransactions: decodedQueuedTransactions, executionStartTimestamp: decodEdexecutionStartTimestamp };
+}
+
+export async function extractTimelockQueuedTransactionsBatchParamsDataAndHash(hre: HardhatRuntimeEnvironment, txHash: string) {
```

Comment:
create type and set return type in this function

Reply:
it's inferred

Reply:
fixed it